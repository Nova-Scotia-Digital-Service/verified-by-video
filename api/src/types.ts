import { Prompt } from '../../shared/types'

export type * from '../../shared/types'

export interface Session {
  version: '1'
  id: string
  createdAt: Date
  expiresAt: Date
  prompts: Prompt[]
}

export interface Submission {
  id: string
  video_url: string
  session: Session
}

export type SubmissionConfirm = Pick<Submission, 'id' | 'video_url'>
