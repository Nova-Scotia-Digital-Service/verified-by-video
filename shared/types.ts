export type ReviewStatus = 'STARTED' | 'ESCALATED' | 'APPROVED' | 'REJECTED'
export type SubmissionStatus = 'NEW' | 'UNDER_REVIEW' | 'ESCALATED' | 'APPROVED' | 'REJECTED'

export type Prompt = {
  id: string
  text: string
  type: 'text'
}

export type Session = {
  version: '1'
  id: string
  created_at: Date
  expires_at: Date
  prompts: Prompt[]
}

export type Submission = {
  id: string
  video_url: string
  upload_date: Date
  session_id: string
}

export type APISubmissionSummary = {
  id: string
  upload_date: Date
  reviews: {
    id: string
    status: ReviewStatus
    created_at: Date
    reviewer_name: string
  }[]
  status: SubmissionStatus
}

export type IdentificationCard = {
  id: string
  session_id: string
  description: string
  photo_url?: string
  upload_date?: Date
}

export type Reviewer = {
  id: string
  email: string
  full_name: string
  is_admin: boolean
}

export type Review = {
  id: string
  status: ReviewStatus
  submission: Submission
  tags: Tag['text'][]
  prompts: Prompt[]
  identification_cards: IdentificationCard[]
  questions: ReviewQuestion[]
  reviewer: Reviewer | null
}

export type ReviewSummary = Omit<Review, 'prompts' | 'identification_cards' | 'questions'>
export type ReviewList = ReviewSummary[]

export type ReviewQuestion = {
  id: string
  question: string
  options: {
    id: string
    text: string
  }[]
}

export type Tag = {
  id: string
  text: string
}
