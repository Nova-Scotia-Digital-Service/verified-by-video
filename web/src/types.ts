export interface ReviewQuestion {
  id: string
  question: string
  options: string[]
}

export type ReviewStatus = 'Unreviewed' | 'Approved' | 'Denied'

export interface VideoData {
  id: string
  upload_date: Date
  status: ReviewStatus
}
