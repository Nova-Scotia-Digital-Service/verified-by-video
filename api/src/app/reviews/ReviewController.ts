import * as TD from '../../types'
import { Get, Controller, Param, UseGuards, Post, Body, Req } from '@nestjs/common'
import { ApiBearerAuth as SwaggerRequireAuth } from '@nestjs/swagger'

import { AuthGuard } from '../auth/AuthGuard'

import { signMinioUrl } from '../../minio'

import { createReview, getReview, finishReview, getConnectionIdForReview } from './ReviewData'

import { offerCredential } from '../../utils/traction'
import { getSubmitterIdentity } from '../../utils/getSubmitterIdentity'
import { credDefId } from '../constants'

@Controller('/staff/reviews')
export class ReviewController {
  @UseGuards(AuthGuard)
  @SwaggerRequireAuth()
  @Post('/')
  public async createReview(
    @Req() request: { user: TD.DBReviewer },
    @Body() body: TD.APICreateReviewRequest,
  ): Promise<TD.APICreateReviewResponse> {
    const { id } = await createReview(request.user, body.submission_id)
    return { id }
  }

  @UseGuards(AuthGuard)
  @SwaggerRequireAuth()
  @Get('/:review_id')
  public async getReview(@Param('review_id') review_id: string): Promise<TD.Review> {
    const { review, questions, prompts, selectedOptions, identificationCards } = await getReview(review_id)

    const reviewQuestions: TD.ReviewQuestion[] = []

    for (const question of questions) {
      const existingQuestion = reviewQuestions.find(({ id }) => question.id === id)
      if (!existingQuestion) {
        const newQuestion = {
          id: question.id,
          question: question.question,
          options: [
            {
              id: question.option_id,
              text: question.option_text,
              selected: selectedOptions?.includes(question.option_id) || false,
              valid: question.option_valid,
            },
          ],
        }
        reviewQuestions.push(newQuestion)
      } else {
        existingQuestion.options.push({
          id: question.option_id,
          text: question.option_text,
          selected: selectedOptions?.includes(question.option_id) || false,
          valid: question.option_valid,
        })
      }
    }

    const signedVideoUrl = await signMinioUrl(review.video_url)

    const signedIdentificationCards = await Promise.all(
      identificationCards.map(async (card) => {
        if (!card.photo_url) return card

        const signedPhotoUrl = await signMinioUrl(card.photo_url)

        return {
          ...card,
          photo_url: signedPhotoUrl,
        }
      }),
    )

    return {
      id: review.id,
      status: review.status,
      submission: {
        id: review.submission_id,
        session_id: review.session_id,
        video_url: signedVideoUrl,
        upload_date: review.upload_date,
        tags: review.tags,
      },
      identification_cards: signedIdentificationCards,
      prompts: prompts.map((prompt) => ({ ...prompt, type: 'text' })),
      questions: reviewQuestions,
      comment: review.comment,
    }
  }

  @UseGuards(AuthGuard)
  @SwaggerRequireAuth()
  @Post('/:review_id')
  public async finishReview(
    @Req() request: { user: TD.DBReviewer },
    @Param('review_id') review_id: string,
    @Body() body: TD.APIFinishReviewRequest,
  ): Promise<void> {
    await finishReview(request.user, review_id, body.status, Object.values(body.answers), body.comment)

    if (body.status === 'APPROVED') {
      const connection_id = await getConnectionIdForReview(review_id)

      if (!connection_id) {
        throw new Error(`No connection id found for review ${review_id}, unable to issue credential`)
        return
      }
      const reviewResponse = await getReview(review_id)
      const response = await getSubmitterIdentity(reviewResponse.review.license_number)
      console.log('Submitter response:', response)

      const credential: TD.PersonCredential = {
        connection_id,
        auto_issue: true,
        auto_remove: true,
        trace: true,
        cred_def_id: credDefId,
        credential_preview: {
          '@type': 'https://didcomm.org/issue-credential/1.0/credential-preview',
          attributes: [
            {
              name: 'family_name',
              value: response.lastName,
            },
            {
              name: 'given_names',
              value: response.firstName,
            },
            {
              name: 'birthdate_dateint',
              value: response.birthdate,
            },
            {
              name: 'street_address',
              value: response.street_address,
            },
            {
              name: 'locality',
              value: response.locality,
            },
            {
              name: 'region',
              value: response.region,
            },
            {
              name: 'postal_code',
              value: response.postal_code,
            },
            {
              name: 'country',
              value: response.country,
            },
            {
              name: 'expiry_date_dateint',
              value: response.expiry_date_dateint,
            },
            {
              name: 'picture',
              value: response.picture,
            },
          ],
        },
        comment: 'Issued by the review',
      }

      await offerCredential(credential)
    }
  }
}
