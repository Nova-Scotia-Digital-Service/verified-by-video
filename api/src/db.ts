import { Pool } from 'pg'

import config from './config'

export const pool = new Pool({
  user: config.PG_USER,
  password: config.PG_PASSWORD,
  host: config.PG_HOST,
  database: config.PG_DATABASE,
})
