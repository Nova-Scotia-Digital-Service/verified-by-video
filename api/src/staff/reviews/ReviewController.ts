import { Get, JsonController, Param } from 'routing-controllers'
import { Service } from 'typedi'

import { mockReviewList, mockReview } from './ReviewData'

@JsonController('/staff/reviews')
@Service()
export class ReviewController {
  @Get('/')
  public async getReviewList() {
    return mockReviewList
  }

  @Get('/:reviewId')
  public async getReview(@Param('reviewId') reviewId: string) {
    return mockReview
  }
}
