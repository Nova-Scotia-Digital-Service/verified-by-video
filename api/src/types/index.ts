export type * from '../../../shared/types'

export type CredentialAttribute = {
  name: string
  value: string
}

export type CredentialPreview = {
  '@type': string
  attributes: [
    CredentialAttribute & { name: 'family_name' },
    CredentialAttribute & { name: 'given_names' },
    CredentialAttribute & { name: 'birthdate_dateint' },
    CredentialAttribute & { name: 'street_address' },
    CredentialAttribute & { name: 'locality' },
    CredentialAttribute & { name: 'region' },
    CredentialAttribute & { name: 'postal_code' },
    CredentialAttribute & { name: 'country' },
    CredentialAttribute & { name: 'expiry_date_dateint' },
    ...(CredentialAttribute & { name: 'picture' })[], // Optional
  ]
}

export type PersonCredential = {
  connection_id: string
  auto_issue: boolean
  auto_remove: boolean
  trace: boolean
  cred_def_id: string
  credential_preview: CredentialPreview
  comment: string
}

export type DrpcSessionRequest = {
  id: number
  jsonrpc: string
  connection_id: string
  thread_id: string
  request: {
    request: {
      method: string
    }
  }
}

export type DrpcResponse = {
  id: number
  jsonrpc: string
  result: any
}

export type DecodedKeycloakToken = {
  sub: string
  email: string
  name: string
  realm_access: {
    roles: ('reviewer' | 'admin')[]
  }
}

export type DBReviewer = {
  id: string
  email: string
  full_name: string
  is_admin: boolean
}

export type Submitter = {
  firstName: string
  lastName: string
  driversLicenseNumber: string
  birthdate: string
}
