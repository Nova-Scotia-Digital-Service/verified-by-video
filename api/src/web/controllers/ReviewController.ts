import { Get, JsonController } from 'routing-controllers'
import { Service } from 'typedi'

import { videoReview } from '../content/VideoData'

@JsonController('/reviews')
@Service()
export class SubmissionController {
  @Get('/')
  public async postSubmissions() {
    return videoReview
  }
}
