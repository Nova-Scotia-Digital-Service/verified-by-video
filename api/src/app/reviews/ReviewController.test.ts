import * as TD from '../../types'

import { pool } from '../../db'
import { unpopulateDb, populateDb } from '../../../test/testUtils'

import { ReviewController } from './ReviewController'

const SIGNED_URL_REGEX = /^https?\:\/\/.*Amz-Signature=[0-9a-z]{64}$/

beforeAll(async () => {
  await populateDb()
})

afterAll(async () => {
  await unpopulateDb()
  await pool.end()
})

// replace variable signed urls in snapshot with constant string
expect.addSnapshotSerializer({
  serialize: () => 'Any<SignedURL>',
  test: (val) => val && typeof val === 'string' && SIGNED_URL_REGEX.test(val),
})

describe('ReviewController', () => {
  describe('getReview', () => {
    it('returns data', async () => {
      expect(await new ReviewController().getReview('532bd3f4-a0c5-4a97-87a9-19d46981747d')).toMatchSnapshot()
      expect(await new ReviewController().getReview('cbb5b46a-ad3b-4a5f-954f-bff18024d1d6')).toMatchSnapshot()
    })
  })

  describe('finishReview', () => {
    it('accepts data', async () => {
      const reviewId = '532bd3f4-a0c5-4a97-87a9-19d46981747d'
      const user: TD.DBUser = {
        id: '837954a6-4ff2-449a-ac0b-35b9883fe1ec',
        email: 'test@example.co.uk',
        full_name: 'Admin User',
        is_admin: true,
      }

      const reviewAnswers = {
        '1ae8b631-a95d-4b55-a599-aec2919096c0': '80ed0dc2-1a93-4c38-ae36-25cbc8659c53',
        '3ea886c1-b97c-4e30-90bd-f1551df1a632': '3b1ca3d6-de80-4b27-aeff-2c8d4237088c',
        '5f816b66-d431-4642-9822-ae2dbd186e38': '7d479637-c890-47fb-b879-a9c27355e776',
        '5e6fd189-c934-4049-bc00-246f2cd9c85e': '4b85d528-a3a8-49c3-a13d-30c566c0c329',
        '9d211ed5-59c4-415d-b782-be6499205b2f': 'a6005f55-aac0-4e26-ba67-11aff456bf5b',
        '676fac34-b6e5-4cff-b523-0a4bbc7bca88': '357a3e84-8376-4594-ab3f-a6b8c436c2da',
        '71712d42-c574-46cf-826e-f76db3a9749f': '86b271cf-6517-40ac-a11b-ae91c81dbafa',
        '957250ed-35cb-4315-a2f5-3efd4667f829': '64b8cfd3-6d98-40c8-af7a-1dc1c9cc4df2',
        '62333928-ade6-41c6-b131-d11efff04179': '153bc35d-3ae4-4746-abd4-106b12aeb187',
      }

      await new ReviewController().finishReview({ user }, reviewId, { status: 'APPROVED', answers: reviewAnswers })
    })
  })
})
