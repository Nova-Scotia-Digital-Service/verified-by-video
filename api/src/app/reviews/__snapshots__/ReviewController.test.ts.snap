// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewController getReview returns data 1`] = `
{
  "comment": null,
  "id": "532bd3f4-a0c5-4a97-87a9-19d46981747d",
  "identification_cards": [
    {
      "description": "Photo from the mobile app",
      "id": "b70fc479-f960-4e43-92b0-669adf225a95",
      "photo_url": Any<SignedURL>,
      "session_id": "88acfa3d-bf2f-4cae-8746-60a9106f6d56",
      "upload_date": 2024-05-08T19:03:51.000Z,
    },
    {
      "description": "Front of Nova Scotia Driver's License",
      "id": "0cf13516-3636-437e-8863-13e7160a7ae9",
      "photo_url": Any<SignedURL>,
      "session_id": "88acfa3d-bf2f-4cae-8746-60a9106f6d56",
      "upload_date": 2024-05-08T19:04:12.000Z,
    },
    {
      "description": "Back of Nova Scotia Driver's License",
      "id": "41cf1ceb-8801-4c88-9e87-cc93bd0f17a5",
      "photo_url": Any<SignedURL>,
      "session_id": "88acfa3d-bf2f-4cae-8746-60a9106f6d56",
      "upload_date": 2024-05-08T19:05:53.000Z,
    },
  ],
  "prompts": [
    {
      "id": "10125e01-642d-4be5-bf1c-2e0575091b23",
      "session_id": "88acfa3d-bf2f-4cae-8746-60a9106f6d56",
      "text": "Hold up two fingers",
      "type": "text",
    },
    {
      "id": "8e579ca4-400f-4802-952a-54ae33e81799",
      "session_id": "88acfa3d-bf2f-4cae-8746-60a9106f6d56",
      "text": "Touch your ear",
      "type": "text",
    },
    {
      "id": "870fbfb4-2bab-4a91-b5b3-0d16d7220353",
      "session_id": "88acfa3d-bf2f-4cae-8746-60a9106f6d56",
      "text": "Turn your head to the left",
      "type": "text",
    },
  ],
  "questions": [
    {
      "id": "62333928-ade6-41c6-b131-d11efff04179",
      "options": [
        {
          "id": "153bc35d-3ae4-4746-abd4-106b12aeb187",
          "selected": false,
          "text": "Yes, all prompts followed by user in the correct order",
          "valid": true,
        },
        {
          "id": "390641de-5288-4261-ad59-7f28b05482c0",
          "selected": false,
          "text": "Didn't follow the prompts correctly",
          "valid": false,
        },
        {
          "id": "287b76d4-21b4-4f5b-bfaa-1163abe5cc42",
          "selected": false,
          "text": "Too low quality video/photo for matching",
          "valid": false,
        },
        {
          "id": "02955e84-81b2-4eb9-9aef-4913ec9e3dc8",
          "selected": false,
          "text": "Couldn't complete for other reason",
          "valid": false,
        },
      ],
      "question": "Did the user correctly follow the prompts in the video?",
    },
    {
      "id": "5f816b66-d431-4642-9822-ae2dbd186e38",
      "options": [
        {
          "id": "7d479637-c890-47fb-b879-a9c27355e776",
          "selected": false,
          "text": "NONPHOTO, PERCY or an acceptable variation",
          "valid": true,
        },
        {
          "id": "b392059a-445a-436b-8fcf-cf43a5b6b8da",
          "selected": false,
          "text": "Didn't provide the correct name",
          "valid": false,
        },
        {
          "id": "d1ce043d-8b6d-42c2-9c03-0972a514c3ca",
          "selected": false,
          "text": "Couldn't complete for other reason",
          "valid": false,
        },
      ],
      "question": "What name did they provide?",
    },
    {
      "id": "9d211ed5-59c4-415d-b782-be6499205b2f",
      "options": [
        {
          "id": "a6005f55-aac0-4e26-ba67-11aff456bf5b",
          "selected": false,
          "text": "Yes, all match",
          "valid": true,
        },
        {
          "id": "0da3c0e3-6535-496b-85cc-186cc84ca767",
          "selected": false,
          "text": "Not all match",
          "valid": false,
        },
        {
          "id": "eb6f5cdf-14fc-455e-bab4-2b27e98942c9",
          "selected": false,
          "text": "Too low quality video or photo for matching",
          "valid": false,
        },
        {
          "id": "6a9a2cb0-b9a2-4959-b483-bbc6c4197593",
          "selected": false,
          "text": "Couldn't complete for other reason",
          "valid": false,
        },
      ],
      "question": "Do all three match: photo on the ID document, video, photo taken in the app?",
    },
    {
      "id": "5e6fd189-c934-4049-bc00-246f2cd9c85e",
      "options": [
        {
          "id": "4b85d528-a3a8-49c3-a13d-30c566c0c329",
          "selected": false,
          "text": "Nova Scotia Driver's License",
          "valid": true,
        },
        {
          "id": "c694c9d7-d5f4-4024-8834-21a800c3a4be",
          "selected": false,
          "text": "Other type",
          "valid": false,
        },
        {
          "id": "ece94423-fb18-4aa9-8c16-6f46da09c259",
          "selected": false,
          "text": "I can't confirm",
          "valid": false,
        },
      ],
      "question": "What type of ID is provided?",
    },
    {
      "id": "957250ed-35cb-4315-a2f5-3efd4667f829",
      "options": [
        {
          "id": "64b8cfd3-6d98-40c8-af7a-1dc1c9cc4df2",
          "selected": false,
          "text": "12121212",
          "valid": true,
        },
        {
          "id": "c40383be-a2c2-4eb7-8b91-93ee430efb3f",
          "selected": false,
          "text": "Other type",
          "valid": false,
        },
        {
          "id": "5254b514-30bf-4615-8db5-b8d1225b4488",
          "selected": false,
          "text": "I can't confirm",
          "valid": false,
        },
      ],
      "question": "What number is on Nova Scotia Driver's License?",
    },
    {
      "id": "3ea886c1-b97c-4e30-90bd-f1551df1a632",
      "options": [
        {
          "id": "3b1ca3d6-de80-4b27-aeff-2c8d4237088c",
          "selected": false,
          "text": "January 1, 1990",
          "valid": true,
        },
        {
          "id": "19a884e4-02d9-4f41-9fb6-2a7b528f3db4",
          "selected": false,
          "text": "Does not match",
          "valid": false,
        },
        {
          "id": "a44ee7e6-ae27-42ef-aa3e-ce9b54a3dbcf",
          "selected": false,
          "text": "I can't confirm",
          "valid": false,
        },
      ],
      "question": "What is the birthdate on the ID?",
    },
    {
      "id": "1ae8b631-a95d-4b55-a599-aec2919096c0",
      "options": [
        {
          "id": "80ed0dc2-1a93-4c38-ae36-25cbc8659c53",
          "selected": false,
          "text": "NONPHOTO, PERCY or an acceptable variation",
          "valid": true,
        },
        {
          "id": "f4b80b72-4df8-4433-bbf6-41fc80ed8455",
          "selected": false,
          "text": "Does not match",
          "valid": false,
        },
        {
          "id": "b87e1c07-0af9-44f1-85df-e32142a8c5d3",
          "selected": false,
          "text": "I can't confirm",
          "valid": false,
        },
      ],
      "question": "What name is on the ID?",
    },
    {
      "id": "71712d42-c574-46cf-826e-f76db3a9749f",
      "options": [
        {
          "id": "86b271cf-6517-40ac-a11b-ae91c81dbafa",
          "selected": false,
          "text": "Is valid, not expired",
          "valid": true,
        },
        {
          "id": "67440bce-44ca-4c57-9d0a-e6e8881a2997",
          "selected": false,
          "text": "Expired",
          "valid": false,
        },
        {
          "id": "c0df2424-327b-4220-b6bd-add4dc58fd92",
          "selected": false,
          "text": "Doesn't appear to be genuine (e.g. not an original document, missing security feature)",
          "valid": false,
        },
        {
          "id": "dbd486a9-e3b5-4fe1-a0ba-0a7682422684",
          "selected": false,
          "text": "Not an acceptable document",
          "valid": false,
        },
      ],
      "question": "Is the ID valid?",
    },
    {
      "id": "676fac34-b6e5-4cff-b523-0a4bbc7bca88",
      "options": [
        {
          "id": "357a3e84-8376-4594-ab3f-a6b8c436c2da",
          "selected": false,
          "text": "Yes I am confident",
          "valid": true,
        },
        {
          "id": "6fea484e-c28e-45b2-ab38-a84a4790eefa",
          "selected": false,
          "text": "Possible suspicious activity",
          "valid": false,
        },
        {
          "id": "d2d4ec17-dff5-4ece-8923-44f6130d195f",
          "selected": false,
          "text": "Not confident enough to verify for other reason",
          "valid": false,
        },
      ],
      "question": "Are you confident this is who they say they are?",
    },
  ],
  "status": "STARTED",
  "submission": {
    "id": "214f747f-778a-4d4b-ab3b-5ed0c9410c69",
    "session_id": "88acfa3d-bf2f-4cae-8746-60a9106f6d56",
    "tags": [
      {
        "id": "67302e01-4197-47e1-b78e-466064ac5691",
        "text": "Tag 2",
      },
      {
        "id": "1529c4ad-c5f7-404c-b880-6ffc9ad4ca1c",
        "text": "Tag 1",
      },
    ],
    "upload_date": 2024-05-08T19:01:48.776Z,
    "video_url": Any<SignedURL>,
  },
}
`;

exports[`ReviewController getReview returns data 2`] = `
{
  "comment": null,
  "id": "cbb5b46a-ad3b-4a5f-954f-bff18024d1d6",
  "identification_cards": [
    {
      "description": "Photo on Nova Scotia Health Card",
      "id": "caa2e19c-0024-4011-bafd-aa6dcdc51b8c",
      "photo_url": null,
      "session_id": "33aeaab8-0a6f-477f-815a-eb0559b1ba3a",
      "upload_date": null,
    },
    {
      "description": "Photo from the mobile app",
      "id": "ec302835-cdcc-459a-bd5f-368b865a6a8a",
      "photo_url": Any<SignedURL>,
      "session_id": "33aeaab8-0a6f-477f-815a-eb0559b1ba3a",
      "upload_date": 2024-05-08T19:03:51.000Z,
    },
  ],
  "prompts": [
    {
      "id": "4233b4fe-1690-4c4e-b6d7-06b3be941be0",
      "session_id": "33aeaab8-0a6f-477f-815a-eb0559b1ba3a",
      "text": "Close your eyes",
      "type": "text",
    },
    {
      "id": "2f5f457c-001c-464d-b224-3eb982af45f7",
      "session_id": "33aeaab8-0a6f-477f-815a-eb0559b1ba3a",
      "text": "Hold up right palm",
      "type": "text",
    },
    {
      "id": "50ff9898-9bba-43be-ab68-8dcb9ec3e879",
      "session_id": "33aeaab8-0a6f-477f-815a-eb0559b1ba3a",
      "text": "Nod your head",
      "type": "text",
    },
  ],
  "questions": [],
  "status": "REJECTED",
  "submission": {
    "id": "be5a2859-8c30-4970-978f-ea30b295ad86",
    "session_id": "33aeaab8-0a6f-477f-815a-eb0559b1ba3a",
    "tags": [],
    "upload_date": 2024-05-10T13:05:18.691Z,
    "video_url": Any<SignedURL>,
  },
}
`;
